src/evm/mod.rs

```diff
@@ -104,25 +105,46 @@ pub fn evm_xcontract_call_with_constructors(
     Ok(true)
 }
 
-pub fn _evm_tx_with_deposit(
+pub fn _evm_run_with_gas_charging(
     log_to: Option<&Path>,
+    charging_policy: Option<(Uint256, Uint256, Uint256)>,
     debug: bool,
     _profile: bool,
 ) -> Result<bool, ethabi::Error> {
+    // returns Ok(true) if success, Ok(false) if insufficient gas money, Err otherwise
     use std::convert::TryFrom;
-    let rt_env = RuntimeEnvironment::new(Uint256::from_usize(1111));
+    let rt_env = RuntimeEnvironment::new(Uint256::from_usize(1111), charging_policy);
     let mut machine = load_from_file(Path::new("arb_os/arbos.mexe"), rt_env);
     machine.start_at_zero();
 
     let my_addr = Uint256::from_usize(1025);
-
-    let mut fib_contract = AbiForContract::new_from_file(&fib_contract_path("Fibonacci"))?;
-    if fib_contract.deploy(&[], &mut machine, Uint256::zero(), None, None, debug) == None {
-        panic!("failed to deploy Fibonacci contract");
+    machine.runtime_env.insert_eth_deposit_message(
+        my_addr.clone(),
+        my_addr.clone(),
+        Uint256::_from_eth(1),
+    );
+    machine.runtime_env.insert_eth_deposit_message(
+        Uint256::from_u64(1025),
+        Uint256::from_u64(1025),
+        Uint256::_from_eth(1),
+    );
```

Comment:
Is there a reason why this is duplicated

Reply:
Looks like an integration hiccup. Removed the duplicate.